testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/FilesystemRealmEncryptedTestCase.java

```diff
@@ -0,0 +1,209 @@
+        private static final String SECRET_KEY = "key";
+        private static final String PREDEFINED_HTTP_SERVER_MECHANISM_FACTORY = "global";
+        private static ManagementClient managementClient;
+        private static boolean badCredentialStore = false;
+
+
+        @Override
+        public void setup(ManagementClient managementClient, java.lang.String s) throws Exception {
+            try (CLIWrapper cli = new CLIWrapper(true)) {
+                cli.sendLine("/subsystem=elytron/simple-role-decoder=from-roles-attribute:add(attribute=Roles)");
+            }
+            setUpTestDomain(DOMAIN_NAME, REALM_NAME, "filesystem", USER, PASSWORD, DEPLOYMENT_ENCRYPTED, CREDENTIAL_STORE, SECRET_KEY);
+            SetUpTask.managementClient = managementClient;
+            ServerReload.reloadIfRequired(managementClient);
+        }
+
+        private void setUpTestDomain(String domainName, String realmName, String path, String username, String password, String deployment, String credentialStore, String secretKey) throws Exception {
+            try (CLIWrapper cli =  new CLIWrapper(true)) {
+                cli.sendLine(String.format("/subsystem=elytron/secret-key-credential-store=%1$s:add(path=%1$s.cs, relative-to=jboss.server.config.dir, create=true, populate=true)",
+                        credentialStore));
+                cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s:add(path=%s,relative-to=jboss.server.config.dir, credential-store=%s, secret-key=%s)",
+                        realmName, path, credentialStore, secretKey));
+                cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s:add-identity(identity=%s)", realmName, username));
+                cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%1$s:set-password(identity=%2$s, digest={algorithm=digest-md5, realm=%1$s, password=%3$s})",
+                        realmName, username, password));
+                cli.sendLine(String.format("/subsystem=elytron/filesystem-realm=%s:add-identity-attribute(identity=%s, name=Roles, value=[JBossAdmin])",
+                        realmName, username));
+                cli.sendLine(String.format("/subsystem=elytron/security-domain=%1$s:add(realms=[{realm=%2$s,role-decoder=from-roles-attribute}],default-realm=%2$s,permission-mapper=default-permission-mapper)",
+                        domainName, realmName));
+                cli.sendLine(String.format(
+                        "/subsystem=elytron/http-authentication-factory=%1$s:add(http-server-mechanism-factory=%2$s,security-domain=%1$s,"
+                                + "mechanism-configurations=[{mechanism-name=BASIC,mechanism-realm-configurations=[{realm-name=\"%1$s\"}]}])",
+                        domainName, PREDEFINED_HTTP_SERVER_MECHANISM_FACTORY));
+                cli.sendLine(String.format(
+                        "/subsystem=undertow/application-security-domain=%s:add(http-authentication-factory=%s)",
```

Comment:
Here instead of `http-authentication-factory=...`, you can just use `security-domain=...`.